ext/otel_observer.c

```diff
@@ -192,6 +298,50 @@ static inline void func_get_lineno(zval *zv, zend_execute_data *ex) {
     }
 }
 
+static inline void func_get_attribute_args(zval *zv, HashTable *attributes,
+                                           zend_execute_data *ex) {
+    if (!OTEL_G(attr_hooks_enabled)) {
+        ZVAL_EMPTY_ARRAY(zv);
+        return;
+    }
+    zend_attribute *attr = find_withspan_attribute(ex->func);
```

Comment:
Can you think of any potential issues with saving this during resolve time in `otel_observer` structure to reduce the extra work here? With multiple potential interfaces it might not be a very cheap operation.

Reply:
I thought that since all interfaces were already available in `func->common.scope` that it would be fairly cheap to process them. I don't see any issues with stashing data in `otel_observer` (I assume you mean as a cache of functions to not re-check).